server/middlewares/index.js

```diff
@@ -1,13 +1,45 @@
-const {decodeToken} = require('../utils');
+const { decodeToken } = require('../utils');
+const {User} = require('../models');
 
-exports.verifyJwt = (req, res, next) => {
-    const {authorization} = req.headers;
-    if (authorization) {
-        const token = authorization.split(' ')[1];
-        const decodedToken = decodeToken(token);
-        req.user = decodedToken.payload;
-        next();
+exports.isAuth = (req, res, next) => {
+  const { authorization } = req.headers;
+  if (authorization) {
+    const token = authorization.split(' ')[1];
+    const { payload } = decodeToken(token);
+    if (payload && payload._id) {
+      User.findById(payload._id)
+        .populate('projects')
+        .exec((err, user) => {
+          if (err || !user) {
+            res.status(400).json({
+              error: 'User not found'
+            });
+          } else {
+            const { _id, name, email, projects } = user;
+            req.profile = { _id, name, email, projects };
+            next();
+          }
+        });
     } else {
-        return res.status(500).send({error: 'not authorized'})
+      return res.status(403).send({ error: 'Access denied' });
     }
-}
\ No newline at end of file
+  } else {
+    return res.status(403).send({ error: 'Access denied' });
+  }
+};
+
+//IS THIS NEEDED BY FUTURE ROUTES?
+// exports.userById = (req, res, next, id) => {
+//   User.findById(id)
+//     .populate('projects')
+//     .exec((err, user) => {
+//       if (err || !user) {
+//         return res.status(400).json({
+//           error: 'User not found'
+//         });
+//       }
+//       const { _id, name, email, projects } = user;
+//       req.profile = { _id, name, email, projects };
+//       next();
+//     });
```

Comment:
I am using this on, PR #42 but we can change the route used by frontend if needed

Reply:
Thanks for the heads up. I'll test it out on a variety of routes to make sure I don't break everything.